src/main/java/listeners/MainEventListener.java

```diff
@@ -0,0 +1,232 @@
+
+        Topic topic = topicOptional.get();
+        if (topic.isInQueue(member)) {
+            topic.removeFromQueue(member);
+            channel.sendMessage(String.format(
+                    "%s has left the \"%s\" queue.",
+                    member.getAsMention(),
+                    args[0])).queue();
+        } else {
+            topic.addToQueue(member);
+            channel.sendMessage(String.format(
+                    "%s has joined the \"%s\" queue.",
+                    member.getAsMention(),
+                    args[0])).queue();
+        }
+    }
+
+    private void ready(Member member, TextChannel channel, Server server, String[] args) {
+        if (!checkAdmin(member, channel)) return;
+
+        Optional<Topic> topicOptional = checkTopicExists(member, channel, server, args[0]);
+        if (topicOptional.isEmpty()) return;
+
+        Topic topic = topicOptional.get();
+        Member mentee = topic.getNextFromQueue();
+        channel.sendMessage(String.format(
+                "%s is ready for %s.",
+                member.getAsMention(),
+                mentee.getAsMention())).queue();
+    }
+
+    private void showqueue(Member member, TextChannel channel, Server server, String[] args) {
+        Optional<Topic> topicOptional = checkTopicExists(member, channel, server, args[0]);
+        if (topicOptional.isEmpty()) return;
+
+        Topic topic = topicOptional.get();
+        if (topic.getMembersInQueue().length == 0) {
+            channel.sendMessage(String.format(
+                    "%s Queue \"%s\" is empty.",
+                    member.getAsMention(),
+                    topic.getName())).queue();
+        } else {
+            String menteeList = Arrays.stream(topic.getMembersInQueue())
+                    .map(Member::getEffectiveName)
+                    .collect(Collectors.joining("\n"));
+
+            channel.sendMessage(String.format(
+                    "%s Members in \"%s\" queue:\n%s",
+                    member.getAsMention(),
+                    topic.getName(),
+                    menteeList)).queue();
+        }
+    }
+
+    private void clear(Member member, TextChannel channel, Server server, String[] args) {
+        if (!checkAdmin(member, channel)) return;
+
+        Optional<Topic> topicOptional = checkTopicExists(member, channel, server, args[0]);
+        if (topicOptional.isEmpty()) return;
+
+        Topic topic = topicOptional.get();
+        Arrays.stream(topic.getMembersInQueue()).forEach(topic::removeFromQueue);
+
+        channel.sendMessage(String.format(
+                "%s has cleared the \"%s\" queue.",
+                member.getAsMention(),
+                topic.getName())).queue();
+    }
+
+    private void unknownCommand(Member member, TextChannel channel, Server server, String[] args) {
+        channel.sendMessage(member.getAsMention() + " Command does not exist! Try $help for a list of valid commands.").queue();
+    }
+}
```

Comment:
Newline @ EOF? Although GitHub sometimes doesn't show it

Reply:
You're right, I forgot to add a newline there. I'll make sure to fix that.

Reply:
Nope, never mind on this one as well. There is definitely a newline here on my end, but GitHub isn't showing it. I'm not sure if there's anything I can do to fix that.

Reply:
No worries; GH complains usually but other times it just shows it, other times it doesn't. 
@natfriedman hire me so i can fix bug thanks

Reply:
Oh wow the @ actually worked